### PR TITLE
Calculate tail probabilities in Windowed Gaussian detector with Q function

### LINKED ISSUE #250: NAB scoring requires NuPIC installed, contrary to README
Hi,

The README says to install _"NuPIC (only required if running the Numenta detector)",_ but:

```
$ python run.py -d null --optimize --score --normalize
Traceback (most recent call last):
  File "run.py", line 179, in <module>
    from nab.detectors.gaussian.windowedGaussian_detector import WindowedGaussianDetector
  File "/Users/felix/Devel/NAB/nab/detectors/gaussian/windowedGaussian_detector.py", line 23, in <module>
    from nupic.algorithms import anomaly_likelihood
ImportError: No module named nupic.algorithms
```

### PR DESCRIPTION
**Pull request by BoltzmannBrain**
Fixes #250 
@subutai please review. PR also includes a little cleanup and a bug fix (in run.py) with the windowed gaussian detector. I'll add the results files after you take a look (adding all of them now makes the GitHub dif page laggy).

### CODE REVIEW

File: nab/detectors/gaussian/windowedGaussian_detector.py [hunk 1]
```diff
@@ -18,9 +18,33 @@
...
+
+  if x < distributionParams["mean"]:
+    # Gaussian is symmetrical around mean, so flip to get the tail probability
+    xp = 2*distributionParams["mean"] - x
+    return 1.0 - normalProbability(xp, distributionParams)
+
+  # Calculate the Q function with the complementary error function, explained
+  # here: http://www.gaussianwaves.com/2012/07/q-function-and-error-functions
+  z = (x - distributionParams["mean"]) / distributionParams["stdev"]
+  return 0.5 * math.erfc(z/math.sqrt(2))
```
Comment (subutai) [inline nab/detectors/gaussian/windowedGaussian_detector.py]: That's nice - didn't know erfc was in math.  Hopefully it is not too slow compared to a lookup table?

File: nab/detectors/gaussian/windowedGaussian_detector.py [hunk 2]
```diff
@@ -53,8 +76,8 @@ def handleRecord(self, inputData):
     anomalyScore = 0.0
     inputValue = inputData["value"]
     if len(self.windowData) > 0:
-      anomalyScore = 1 - anomaly_likelihood.normalProbability(
-        inputValue, {"mean": self.mean,"stdev": self.std})
+      anomalyScore = 1 - normalProbability(
+          inputValue, {"mean": self.mean, "stdev": self.std})
```
Comment (subutai) [inline nab/detectors/gaussian/windowedGaussian_detector.py]: No need to pass in a dict - just pass in self.mean and self.std and redefine normalProbability() accordingly.

File: run.py [hunk 3]
```diff
@@ -184,7 +184,8 @@ def main(args):
   if "skyline" in args.detectors:
     from nab.detectors.skyline.skyline_detector import SkylineDetector
   if "windowedGaussian" in args.detectors:
-    import WindowedGaussianDetector
+    from nab.detectors.gaussian.windowedGaussian_detector import (
```
Comment (subutai) [inline run.py]: Might need to merge in master?

**Comment by subutai** [comment]
:+1 Other than comments.

File: run.py (hunk 3 above)
Comment (BoltzmannBrain) [inline run.py]: Yes, I made the change here before making it in #252

File: nab/detectors/gaussian/windowedGaussian_detector.py (hunk 2 above)
Comment (BoltzmannBrain) [inline nab/detectors/gaussian/windowedGaussian_detector.py]: Thank you

File: nab/detectors/gaussian/windowedGaussian_detector.py (hunk 1 above)
Comment (BoltzmannBrain) [inline nab/detectors/gaussian/windowedGaussian_detector.py]: Runs really quick

**Pull request merged**